src/dnx/dnx.cpp

```diff
@@ -1,97 +1,107 @@
-// Defines the entry point for the console application.
-
 #include "stdafx.h"
 #include "dnx.h"
+#include "pal.h"
 
-void GetModuleDirectory(HMODULE module, LPWSTR szPath)
-{
-    DWORD dirLength = GetModuleFileName(module, szPath, MAX_PATH);
-    for (dirLength--; dirLength >= 0 && szPath[dirLength] != '\\'; dirLength--);
-    szPath[dirLength + 1] = L'\0';
-}
-
-int LastIndexOfChar(LPCWSTR const pszStr, WCHAR c)
+bool LastIndexOfChar(LPCTSTR const pszStr, TCHAR c, size_t* pIndex)
 {
-    int nIndex = wcsnlen_s(pszStr, MAX_PATH) - 1;
-    do
+    size_t nIndex = _tcsnlen(pszStr, MAX_PATH) - 1;
```

Comment:
The use of MAX_PATH seems really odd here. There's nothing in the function name to suggest this is dealing with paths.